test/aerosol_activation_calibration.jl

```diff
@@ -0,0 +1,179 @@
+        url = "https://www.dropbox.com/scl/fi/qgq6ujvqenebjkskqvht5/2modal_dataset1_train.csv?rlkey=53qtqz0mtce993gy5jtnpdfz5&dl=0"
+        download(url, fname)
+        mkdir(joinpath(fpath, "data"))
+        mv(fname, joinpath(fpath, "data", fname), force = true)
+    end
+    X_train, Y_train, initial_data =
+        read_aerosol_dataset(joinpath(fpath, "data", fname))
+
+    rng = Random.MersenneTwister(1)
+
+    prior = combine_distributions([
+        constrained_gaussian("f_coeff_1_ARG2000", 0.5, 0.5, 0, Inf),
+        constrained_gaussian("f_coeff_2_ARG2000", 2.5, 0.5, 0, Inf),
+        constrained_gaussian("g_coeff_1_ARG2000", 1.0, 0.5, 0, Inf),
+        constrained_gaussian("g_coeff_2_ARG2000", 0.25, 0.5, 0, Inf),
+        constrained_gaussian("pow_1_ARG2000", 1.5, 0.5, 0, Inf),
+        constrained_gaussian("pow_2_ARG2000", 0.75, 0.5, 0, Inf),
+    ])
+
+    all_params_means = []
+    all_mean_error_metrics = []
+
+    for i in 1:N_samples
+        sample_inds =
+            StatsBase.sample(1:DF.nrow(X_train), sample_size, replace = false)
+        X_sample = X_train[sample_inds, :]
+        Y_sample = Y_train[sample_inds]
+
+        initial_ensemble =
+            EKP.construct_initial_ensemble(rng, prior, N_ensemble)
+        ensemble_kalman_process = EKP.EnsembleKalmanProcess(
+            initial_ensemble,
+            Y_sample,
+            1.0 * EKP.I,
+            EKP.Inversion(),
+        )
+
+        mean_error_metrics = []
+
+        for j in 1:N_iterations_per_sample
+            params_cur = EKP.get_ϕ_final(prior, ensemble_kalman_process)
+            errs = calibration_error_metrics(
+                X_sample,
+                Y_sample,
+                params_cur,
+                aip,
+                tps,
+                FT,
+            )
+            push!(mean_error_metrics, StatsBase.mean(errs))
+
+            pred_ens = hcat(
+                [
+                    calibrated_prediction(
+                        X_sample,
+                        params_cur[:, k],
+                        aip,
+                        tps,
+                        FT,
+                    ) for k in 1:N_ensemble
+                ]...,
+            )
+
+            EKP.update_ensemble!(ensemble_kalman_process, pred_ens)
+        end
+
+        params_final = EKP.get_ϕ_final(prior, ensemble_kalman_process)
+        errs = calibration_error_metrics(
+            X_sample,
+            Y_sample,
+            params_final,
+            aip,
+            tps,
+            FT,
+        )
+        push!(mean_error_metrics, StatsBase.mean(errs))
+
+        params_mean_final = EKP.get_ϕ_mean_final(prior, ensemble_kalman_process)
+        push!(all_params_means, params_mean_final)
+        push!(all_mean_error_metrics, mean_error_metrics)
+    end
+    final_params_means = StatsBase.mean(all_params_means)
+
+    return final_params_means, all_mean_error_metrics
+end
+
+function test_emulator(
+    FT;
+    ml_model = "NN",
```

Comment:
We only test the `calibrated` `ml_model` here. So maybe we don't need this argument and the following `if` statement?